### PR TITLE
Create twitch streamer sidebar

### PR DESCRIPTION
**Pull request by anderssonw**
POC Sidebar for Twitch Streamers

Currently using the streamer prop from the Cookbook model.

Want to add features where you get the live status such as in the old cookbook, but just creating this for the time being. Added it next to sections and currently just residing alone in `[cookbookId]/index`

image

### CODE REVIEW

File: src/components/Sidebar/TwitchSidebar/TwitchSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  streams: Array<string>;
+}
+
+export interface StreamItemProps {
+  stream: string;
+  onlineStatus: string;
+}
+
+function capitalize(s) {
+  return s && s[0].toUpperCase() + s.slice(1);
```
Comment (bartdebever) [inline src/components/Sidebar/TwitchSidebar/TwitchSidebar.tsx]: If you are only using this for the avatar, wouldn't t be easier to only return the first character as upper case? The rest isn't getting used and pretty sure twitch won't allow names with spaces

File: src/pages/cookbooks/[cookbookId]/index.tsx [hunk 2]
```diff
@@ -3,24 +3,39 @@ import { useRouter } from 'next/router';
...
+      if (cookbookId != null) {
+        const cookbook = useStore
+          .getState()
+          .cookbooks.find(cookbook => cookbook._id === cookbookId);
+
+        if (cookbook) {
+          setCookbook(cookbook);
+          const guides = await HttpService.get(ROUTES.GUIDES(cookbook._id));
+          useStore.setState({ cookbook, guides });
+          setstreams(cookbook.streams);
```
Comment (bartdebever) [inline src/pages/cookbooks/[cookbookId]/index.tsx]: I would assume this to be `setStream` instead of ` setstream`

Comment (bartdebever) [inline src/pages/cookbooks/[cookbookId]/index.tsx]: If I get this right, this method locally gets the cookbook from the store, sets it and its streams in the store and then uses the store to pass the streams to the sidebar. Why not just grab it and store it locally?

Comment (anderssonw) [reply src/pages/cookbooks/[cookbookId]/index.tsx]: was just poorly named function in the useState, but yeah i renamed it

File: src/components/Sidebar/TwitchSidebar/TwitchSidebar.tsx (hunk 1 above)
Comment (anderssonw) [reply src/components/Sidebar/TwitchSidebar/TwitchSidebar.tsx]: word

File: src/pages/cookbooks/[cookbookId]/index.tsx (hunk 2 above)
Comment (anderssonw) [reply src/pages/cookbooks/[cookbookId]/index.tsx]: not sure i follow here. i'm not storing streams in the store, it's just a useState hook for setting streams in this component, so thats a local store, no?

as this is the cookbook/id page we should fetch the cookbook from the store based on the route ID, and then we locally store the streams with that cookbook. should i do it differently?

Comment (bartdebever) [reply src/pages/cookbooks/[cookbookId]/index.tsx]: Woops, looked to me like the setStreams was updating the store but it doesn't I guess, thats my bad

Comment (anderssonw) [reply src/pages/cookbooks/[cookbookId]/index.tsx]: alright np!

**Pull request closed without merging**